src/treemodel.cpp

```diff
@@ -637,6 +637,18 @@ void TreeModel::GetFlatJson(const QString& key, const QJsonValue& value, QVector
             GetFlatJson(itemKey, itemValue, stringList);
         }
     }
+    else if (value.isBool())
+    {
+        stringList.append(KeyValueString(key, QString(value.toBool() ? "true" : "false")));
+    }
+    else if (value.isNull())
```

Comment:
I’d rather put empty string for null. 
When can we get undefined?

Reply:
I would prefered "null" over an empty string, since otherwise, it would not be possible to distinguish null from an actually empty string (yes, we have both in hyper...)
Why would you prefer an empty string over null?

Regarding undefined: From http://doc.qt.io/qt-5/qjsonvalue.html#Type-enum states

> The value is undefined. This is usually returned as an error condition, when trying to read an out of bounds value in an array or a non existent key in an object.

I don't think we will ever need to handle it, but better safe than sorry.

Reply:
I have no particular reason for my preference so go ahead and pull. 

I guess I just want a way to distinguish between null that TLV generated and null that might come from the log file itself.

Reply:
I think that this change get tries to fix some of the missed information/ambiguity.
Right now it is not possible to distinguish between null and "null" (the string with the word null in it). Same with true, false and undefined. Perhaps we should attempt to improve the overall formatting later.
Our current format is almost fine, as we were trying to optimize for the most common scenario (displaying just a few key-value pairs with strings in a single line). But it is kind of inefficient with complex objects (like the qp-query-batch event that, when flattened, gets confusing) or ambiguous situations.
Maybe we should revisit how we format stuff. Optimize the format in the main viewer to display on a single line and optimize the format in the value window to be very readable even in complex nested objects.